src/driver/Program.fs

```diff
@@ -668,13 +668,19 @@ let main argv =
                         else
                             return None
                     }
+                    let exitCode =                         
+                        if result.ExitCode = 0 && result.StandardError.Length <> 0 then
+                            -1
```

Comment:
> -1 [](start = 27, length = 3)

Hi Richard, this seems confusing.  Does the engine swallow an exception and simply print the output to stderr?  If so, this should be an engine fix.  If not, what is the code path where this came up?   This change looks good to go in, this is just a clarification question.

Reply:
There was a crash in the fuzzing thread, which aborted that thread and printed to stderr.  When this happens the GC thread then cleans up and the main thread completes. This causes the engine to ultimately return a successful status code. 

The engine could (and probably should) keep track of thread exceptions and, if there were any present, return a non-successful error code. This quick fix seemed appropriate now because the driver probably still should report a failure if there was something in stderr for some reason.

I can create an issue for an engine fix to cure the root problem.

Reply:
Awesome, thank you, this sounds good.


---
In reply to: [541285192](https://github.com/microsoft/restler-fuzzer/pull/109#discussion_r541285192) [](ancestors = 541285192)